cip-TOKEN-STD/cip-meiersi-Canton-Network-Token-Standard.md

```diff
@@ -0,0 +1,567 @@
+
+1. Querying any Scan URL of an SV node for the CNS entry of `/v0/ans-entries/by-party/<admin-party-id>`
+2. Parsing the JSON object in the `description` field of the response as a record.
+3. Accessing the `meta` field of that record to get the metadata entries.
+
+If any of the steps fail, then there is no metadata associated with the `admin` party.
+
+Note that [`/v0/ans-entries/by-party/`](https://docs.dev.sync.global/app_dev/scan_api/scan_openapi.html#get--v0-ans-entries-by-party-party)
+returns the CNS entry for the party with the lexicographically smallest name.
+Registry operators must make sure to store their metadata on that entry for the above procedure to work.
+The easiest way to do so is to allocate at most one CNS entry for their `admin` party.
+
+
+#### Transaction History
+
+The standard aims to enable wallet clients to serve a transaction history view to their users that explains
+all changes to a users' portfolio view and their view of in-progress transfers.
+
+The standard APIs thus define choices for common actions taken by registries,
+wallets, or apps on user-visible contracts that implement one of the standard's
+Daml interfaces.  Thereby enabling wallets to parse the transaction history in
+an implementation-agnostic way using an interface filter to subscribe to the
+transaction tree stream served on the Ledger API of the validator node hosting
+the user's Daml parties.
+
+Wallet clients should implement a fallback mechanism to display changes to the
+user-visible contracts via registry-specific choices that are not part of the
+standard. For example, they could show the JSON rendering of the choice name and
+argument, and offer an option to the user to inspect the full sub-transaction
+below the choice.
+
+
+## Rationale
+
+### Alternatives Considered
+
+#### Shipping UTXOs On-Ledger
+
+As explained in [UTXO Access Management](#utxo-access-management) the standard expects registries
+to serve HTTP APIs that provide access to the UTXO's (i.e., Daml contracts)
+required to call choices on the standard's Daml interfaces.
+In contrast, public chains ship all required UTXOs on-ledger.
+They also do not require any access management for reading data, which they can
+get away with as they make all data public.
+
+Explicitly managing that access is a hard requirement for any ledger with privacy.
+However, one might wonder why we don't ship the data on-ledger for the sake of simplicity.
+We propose to not do so and ship this data via off-ledger via HTTP APIs for the following reasons:
+
+- *network scalability*: all transaction commits must go through a synchronizer, which
+  is thus prone to become a bottleneck. Shipping data on-ledger would imply shipping them via
+  a synchronizer that all relevant participants are connected to, which would increase the chance
+  of it becoming a bottleneck.
+- *hide implementation details*: what UTXO's to fetch for exercising an
```

Comment:
```suggestion
- *traffic costs*: On the global synchronizer, transactions incur traffic costs. Distributing data off-ledger means instead of paying traffic costs on the global synchronizer you only pay for network egress costs for an HTTP API which is usually significantly cheaper.
- *hide implementation details*: what UTXO's to fetch for exercising an
```